src/Evolution/Systems/Cce/WorldtubeDataManager.cpp

```diff
@@ -29,28 +29,37 @@ MetricWorldtubeDataManager::MetricWorldtubeDataManager(
         buffer_updater,
     const size_t l_max, const size_t buffer_depth,
     std::unique_ptr<intrp::SpanInterpolator> interpolator,
-    const bool fix_spec_normalization) noexcept
+    const bool fix_spec_normalization,
+    const std::optional<double> extraction_radius) noexcept
     : buffer_updater_{std::move(buffer_updater)},
       l_max_{l_max},
       fix_spec_normalization_{fix_spec_normalization},
       interpolated_coefficients_{
           Spectral::Swsh::size_of_libsharp_coefficient_vector(l_max)},
       buffer_depth_{buffer_depth},
-      interpolator_{std::move(interpolator)} {
-  if (UNLIKELY(buffer_updater_->get_time_buffer().size() <
-               2 * interpolator_->required_number_of_points_before_and_after() +
-                   buffer_depth)) {
+      interpolator_{std::move(interpolator)},
+      extraction_radius_{extraction_radius} {
+  if (UNLIKELY(
+          buffer_updater_->get_time_buffer().size() <
+          2 * interpolator_->required_number_of_points_before_and_after())) {
     ERROR(
         "The specified buffer updater doesn't have enough time points to "
-        "supply the requested interpolation buffer. This almost certainly "
+        "supply the requested interpolator. This almost certainly "
         "indicates that the corresponding file hasn't been created properly, "
-        "but might indicate that the `buffer_depth` template parameter is "
-        "too large or the specified Interpolator requests too many points");
+        "but might indicate that the specified Interpolator requests too many "
+        "points");
+  }
+  if (UNLIKELY(buffer_updater_->get_time_buffer().size() <
+               2 * interpolator_->required_number_of_points_before_and_after() +
+                   buffer_depth_)) {
+    buffer_depth_ =
+        buffer_updater_->get_time_buffer().size() -
+        2 * interpolator_->required_number_of_points_before_and_after();
```

Comment:
I don't quite understand why you can make this change, can you explain (and perhaps add a comment and/or docs)? It looks like the constructed object may end up with a different `buffer_depth_` than is passed to the constructor, which seems unintuitive to me.

Reply:
Yea, that's intentional (I'll add a code comment). The point is that the buffer depth must not be so large that the worldtube data cannot fill the buffer, so if the input file requests the buffer be larger than the input file size, this adjusts it to be a buffer of length equal to the input file.